packages/core/src/components/essentials/legend.ts

```diff
@@ -80,6 +87,15 @@ export class Legend extends Component {
 			.attr("height", checkboxRadius * 2)
 			.attr("rx", 1)
 			.attr("ry", 1)
+			.attr("class", (d, i) => {
+				if (paletteIndex) {
+					return this.model.getColorClassName(
+						["fill"],
+						d.name,
+						"checkbox"
+					);
+				}
+			})
 			.style("fill", (d) => {
```

Comment:
so we need both `class` & `fill`?

Reply:
we need `class` for color palette and `fill` for custom colors when user provides the color scale.